### ISSUE TITLE
Upgrade to springboot 3.0.5

### ISSUE THREAD

**Issue by craph** [hendisantika/springboot-adminlte3-template#5]
Hello,

Is it possible to upgrade your code with springboot in version `3.0.5` ?

Thank you very much.
Best regards,

**Comment by hendisantika** [comment]
Yes, it is. I will upgrade it now.

**Comment by craph** [comment]
Do you know what is the difference between with securityMatcher and requestMatcher in SecurityFilterChain  ?

**Comment by hendisantika** [comment]
In the Spring Security framework, the `SecurityFilterChain` is responsible for applying a series of security filters to incoming HTTP requests. Two important interfaces that are used to configure a `SecurityFilterChain` are `RequestMatcher` and `SecurityMatcher`.

`RequestMatcher` is an interface that is used to match incoming HTTP requests against a predefined set of criteria. It is used to determine if a particular request should be processed by a particular `SecurityFilterChain`. For example, you could define a `RequestMatcher` that matches all HTTP requests that start with "/api". When a request comes in, Spring Security will use the `RequestMatcher` to determine if it should be processed by the `SecurityFilterChain` that matches that criteria.

`SecurityMatcher` is an interface that is used to match the authentication and authorization requirements for a particular request. It is used to determine if a particular request requires authentication or authorization. For example, you could define a `SecurityMatcher` that requires all requests to be authenticated with a specific set of credentials before they are processed by the `SecurityFilterChain`.

So, to summarize, `RequestMatcher` is used to match incoming requests against a predefined set of criteria, while `SecurityMatcher` is used to match the authentication and authorization requirements for a particular request. Both interfaces are used together to determine which `SecurityFilterChain` should be applied to a particular request.

**Comment by craph** [comment]
Thank you very much for all the information.
Do you have a link that explains all this ?

**Comment by hendisantika** [comment]
For now, just follow. You will understand eventually when you have many experiences.

**Comment by craph** [comment]
Thank you very much

**Comment by hendisantika** [comment]
You're welcome.

**Comment by craph** [comment]
Last question, when you say 

> So, to summarize, RequestMatcher is used to match incoming requests against a predefined set of criteria, while SecurityMatcher is used to match the authentication and authorization requirements for a particular request. Both interfaces are used together to determine which SecurityFilterChain should be applied to a particular request.

SecurityMatcher is also used to match incoming requests no ?

because 
```
@Bean
    SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
        http
            .authorizeHttpRequests((authz) -> authz
                .requestMatchers("/admin/**").hasRole("ADMIN")
                .anyRequest().authenticated()
            );
        return http.build();
    }
```
```
@Bean
public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
    http
        .securityMatcher("/api/**", "/app/**")
        .authorizeHttpRequests((authz) -> authz
            .requestMatchers("/api/admin/**").hasRole("ADMIN")
            .anyRequest().authenticated()
        );
    return http.build();
}
```
both verify `authenticated` right ?

**Comment by hendisantika** [comment]
It is better to user `requestMatchers` if it is only request without security.

**Comment by craph** [comment]
> It is better to user `requestMatchers` if it is only request without security.

but authenticated is security too no ? it's weird

**Comment by hendisantika** [comment]
No, it's different.

**Comment by craph** [comment]
which line of code define security ?
